src/intp/stupid.cpp

```
@@ -1536,6 +1860,17 @@ namespace NG::intp
     void visit(CompoundStatement *stmt) override
     {
       auto blockScopes = fork_scope_chain(activeScopes);
+      auto blockScopeId = current_scope_id(blockScopes);
+      struct ScopeDropGuard
+      {
+        NGSymbols symbols;
+        RuntimeRef<Vec<CallFrame>> frames;
+        uint64_t scopeId = 0;
+        ~ScopeDropGuard()
+        {
+          drop_scope_cells(symbols, frames, scopeId);
```

Cppcheck (2.20.0)

[error] Unhandled exception thrown in function declared not to throw exceptions.

(throwInNoexceptFunction)